community/contributing/updating_the_class_reference.rst

```diff
@@ -1,151 +1,176 @@
 .. _doc_updating_the_class_reference:
 
-Contributing to the class reference
-===================================
+Contributing to the class reference with Git
+============================================
 
 .. highlight:: shell
 
-.. note:: This guide also is available as a `video tutorial on YouTube <https://www.youtube.com/watch?v=5jeHXxeX-JY>`_.
+This page gives you an overview of the steps to submit changes to Godot's class
+reference using the Git version control system.
 
-Godot ships with many nodes and singletons to help you develop your games. Each is a class, documented in the :ref:`class reference <toc-class-ref>`.
-This reference is essential for anyone learning the engine: it is available both online and in the engine.
+The class reference is available online in the :ref:`classes <toc-class-ref>`
+section of the documentation and in the Godot editor, from the help menu.
 
-But it's incomplete. Some methods, variables and signals lack descriptions. Others changed with recent releases and need updates.
-The developers can't write the entire reference on their own. Godot needs you, and all of us, to contribute.
+.. seealso::
+
+    This guide focuses on using Git. You can find the writing guidelines for the
+    class reference :ref:`here <doc_class_reference_writing_guidelines>`.
+
+    If you want to translate the class reference from English to another
+    language, see :ref:`doc_editor_and_docs_localization`.
 
-**Important:** If you are planning to make larger changes or a more substantial contribution, it is usually a good idea
-to create an issue (or a comment in an existing one) to let others know so they don't start working on the same thing too. Class
-reference issues should be opened in the `godot-docs repository <https://github.com/godotengine/godot-docs/issues>`__.
+In the class reference, some methods, variables, and signals lack descriptions.
+Others changed with recent releases and need updates. The developers can't write
+the entire reference on their own. Godot needs you, and all of us, to
+contribute.
+
+**Important:** If you plan to make large changes, you should create an issue on
+ the `godot-docs repository <https://github.com/godotengine/godot-docs/>`_
+ or comment on an existing issue. Doing so lets others know you're already
+ taking care of a given class.
 
 .. seealso::
 
-    Not sure where to start contributing? Take a look at the current class reference
+    Not sure which class to contribute to? Take a look at the class reference's
     completion status `here <https://godotengine.github.io/doc-status/>`__.
 
 How to contribute
 -----------------
 
-The class reference lies in the following XML files, in Godot's GitHub repository: `doc/classes/ <https://github.com/godotengine/godot/tree/master/doc/classes>`_.
+You can find the source files for the class reference in Godot's GitHub
+repository: `doc/classes/
+<https://github.com/godotengine/godot/tree/master/doc/classes>`_.
```

Comment:
As mentioned in another page, this should also call out the fact that some pages are in `modules/<name>/doc_classes/` when they are for module-specific classes.

For the reference:
```
$ ls modules/*/doc_classes/
modules/csg/doc_classes/:
CSGBox3D.xml  CSGCombiner3D.xml  CSGCylinder3D.xml  CSGMesh3D.xml  CSGPolygon3D.xml  CSGPrimitive3D.xml  CSGShape3D.xml  CSGSphere3D.xml  CSGTorus3D.xml

modules/enet/doc_classes/:
NetworkedMultiplayerENet.xml

modules/gdnative/doc_classes/:
GDNativeLibrary.xml  MultiplayerPeerGDNative.xml  PacketPeerGDNative.xml  StreamPeerGDNative.xml   WebRTCDataChannelGDNative.xml     XRInterfaceGDNative.xml
GDNative.xml         NativeScript.xml             PluginScript.xml        VideoStreamGDNative.xml  WebRTCPeerConnectionGDNative.xml

modules/gdscript/doc_classes/:
 GDScriptFunctionState.xml  '@GDScript.xml'   GDScript.xml

modules/gridmap/doc_classes/:
GridMap.xml  README.md

modules/mobile_vr/doc_classes/:
MobileVRInterface.xml

modules/mono/doc_classes/:
CSharpScript.xml  GodotSharp.xml

modules/opensimplex/doc_classes/:
NoiseTexture.xml  OpenSimplexNoise.xml

modules/regex/doc_classes/:
RegExMatch.xml  RegEx.xml

modules/stb_vorbis/doc_classes/:
AudioStreamOGGVorbis.xml

modules/theora/doc_classes/:
VideoStreamTheora.xml

modules/upnp/doc_classes/:
UPNPDevice.xml  UPNP.xml

modules/visual_script/doc_classes/:
VisualScriptBasicTypeConstant.xml  VisualScriptCustomNode.xml       VisualScriptFunction.xml        VisualScriptLocalVar.xml      VisualScriptReturn.xml     VisualScriptTypeCast.xml
VisualScriptBuiltinFunc.xml        VisualScriptDeconstruct.xml      VisualScriptGlobalConstant.xml  VisualScriptMathConstant.xml  VisualScriptSceneNode.xml  VisualScriptVariableGet.xml
VisualScriptClassConstant.xml      VisualScriptEditor.xml           VisualScriptIndexGet.xml        VisualScriptNode.xml          VisualScriptSceneTree.xml  VisualScriptVariableSet.xml
VisualScriptComment.xml            VisualScriptEmitSignal.xml       VisualScriptIndexSet.xml        VisualScriptOperator.xml      VisualScriptSelect.xml     VisualScriptWhile.xml
VisualScriptComposeArray.xml       VisualScriptEngineSingleton.xml  VisualScriptInputAction.xml     VisualScriptPreload.xml       VisualScriptSelf.xml       VisualScript.xml
VisualScriptCondition.xml          VisualScriptExpression.xml       VisualScriptIterator.xml        VisualScriptPropertyGet.xml   VisualScriptSequence.xml   VisualScriptYieldSignal.xml
VisualScriptConstant.xml           VisualScriptFunctionCall.xml     VisualScriptLists.xml           VisualScriptPropertySet.xml   VisualScriptSubCall.xml    VisualScriptYield.xml
VisualScriptConstructor.xml        VisualScriptFunctionState.xml    VisualScriptLocalVarSet.xml     VisualScriptResourcePath.xml  VisualScriptSwitch.xml

modules/webm/doc_classes/:
VideoStreamWebm.xml

modules/webrtc/doc_classes/:
WebRTCDataChannel.xml  WebRTCMultiplayer.xml  WebRTCPeerConnection.xml

modules/websocket/doc_classes/:
WebSocketClient.xml  WebSocketMultiplayerPeer.xml  WebSocketPeer.xml  WebSocketServer.xml
```
(Of course those shouldn't be listed but in case you want to give a couple examples.)

Reply:
@Calinou Maybe we can modify https://godotengine.github.io/doc-status/ (and thus `doc_status.py`) to include a link to the XML file for each class? This way users don't have to do detective work and we can refer to that page as the way to locate XML files.